### DISCUSSION TITLE
SPLIB Library Issues

### DISCUSSION THREAD

**Discussion by Aaron-Amunix** [vdemichev/DiaNN#200] [Q&A] [answered]
Hi Vadim, 

First off thanks for making such an amazing software, it is a pleasure to use/does a great job with library free analysis.    

I have a unique dataset where I am testing various proteases and synthetic substrates (19aa's each) and I want to get an idea of where they cleave. I'm having great success when looking for the full length substrates and their expected fragments (I've created a FASTA file with a unique entry for the full length substrate as well as a separate entry for its expected C-product fragment.

However, I would also like to look for situations where proteases cleave substrates at an unexpected location and/or where there is a truncation during synthesis.  To do this, I've created a .splib library (MsFragger/TPP/Spectrast built) which was run via a non-specific enzyme search.  

However, when I try to analyze my data in DIA-NN, none of the truncated substrates or products of protease cleavage in an unexpected location have any identifications associated with them in the .pr_matrix or the raw data output.  

[example.xlsx](https://github.com/vdemichev/DiaNN/files/7269099/example.xlsx)

I tried this again with a new library searched with non-specific enzyme and only the full length substrates as a unique fasta entries and had the same result.  

In the log, I did notice this which may explain the issue.

[0:00] WARNING: no protein information found in the library. Annotating library precursors with information from the FASTA database
[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete)

I had DIA-NN generate a library and as I anticipated there wasn't any protein annotation on anything that wasn't a complete substrate.  I'm thinking this might be a SPLIB issue and/or my peptides not being tryptic?  All of the peptides are annotated in the .splib and .sptxt files correctly.

Let me know if you would like more information/have any ideas on how to get around it.

Thanks for your help!

-Aaron

**Comment by vdemichev** [comment, marked as answer]
Hi Aaron,

If your experiment is not quant-focused, I would suggest just running DDA + FragPipe.
Specifically about the DIA-NN output. If it seems it's incorrect, i.e. something should be detected but is not, can you please share the log, the spectral library you use, the output, and indicate a peptide which is not being detected?

FragPipe can create libraries in .tsv format, which is compatible with DIA-NN. Please only use the .tsv lib created. Maybe that's the issue?

If you have DIA-NN digest the FASTA in silico, it does so with the indicated enzyme specificity. By default it's trypsin. You can change it but it's impossible to specify a non-specific digest (and I don't recommend doing it using DIA anyway).

Best,
Vadim

**Reply by Aaron-Amunix** [reply]
Hi Vadim, Thanks for the prompt reply!  

My experiment is quantitative/was build for DIA using 7da windows (no co-eluting precursors are within 21 daltons of eachother).  I'm very happy with the in-silico library results I'm getting.  

I am actually detecting many of the the peptides from my DDA library but they not being assigned a Protein Name (instead it is blank).  This is the case in the DIA-NN generated .tsv library too hence why I'm thinking it might just be something to how the .splib/sptxt is imported.  

I'm using MSFragger via TPP, I'll give it a try using Fragpipe generated DIA library and see how that looks.  

Agree in-silico DIA analysis isn't designed for a no-enzyme search.  However I found a unique workaround since I know what enzymatic products I am looking for (more of a very large scale MRM vs a traditional DIA experiment).  What I did to get around this was to use a --cut C* command (I have no cystines in any of my peptides!) and put a separate ID's in my fasta file for the full length substrate and enzymatic products. That search is working perfectly and I'm not having any issues!

**Reply by vdemichev** [reply]
OK, I got it. The enzyme specificity in DIA-NN when digesting the library in silico or using 'Reannotate' must match the way peptides have been obtained, otherwise annotation (of peptides with protein info) will not work. If impossible to specify such an enzyme, and don't want to just use unspecific digest (unspecific might work for 'Reannotate'), then the only solution would be to use some third-party R/Python scripts for annotation.

--cut C* allows all cuts after C but no other cuts. To disable all cuts can just use "--cut " (without quotes), i.e. just leave the arguments of the --cut command empty. If you'd like to enable non-specific, can use --cut G*,A*,L*,I*,etc - just list all AAs. But this is OK for 'Reannotate' only.

Best,
Vadim

**Reply by Aaron-Amunix** [reply]
Makes sense.  What is strange is in my .splib and .sptxt all of the peptides are annotated with their corresponding proteins but that information isn't being recognized by DIA-NN.  

[0:00] WARNING: no protein information found in the library. Annotating library precursors with information from the FASTA database
[0:00] Finding proteotypic peptides (assuming that the list of UniProt ids provided for each peptide is complete) 

I'll try FragPipe and see if a FragPipe generated library will work better and won't need to have proteins re-annotated.  MSFragger can tell what the proteins are but that data just isn't being passed on to DIA-NN using TPP/Spectrast.  

-Aaron

**Reply by Aaron-Amunix** [reply]
Hi Vadim,

Update from me.  Looks like a Fragpipe based .txt library did the trick and fixes this problem.  Thanks for the help/I'll use Fragpipe for this moving forward.  

-Aaron
